### ISSUE TITLE
Domain MX check should make sure MX domain resolves to Modoboa instance

### ISSUE THREAD

**Issue by brylie** [modoboa/modoboa#1143]
# Impacted versions

* Modoboa: 1.7.4
* installer used: Yes

# Steps to reproduce
1. add domain in Modoboa
2. configure MX for domain on DNS provider
    - MX should resolve to different server than Modoboa

# Current behavior
Modoboa reports that the MX settings are 'green' (i.e. OK) when, in fact, the MX settings resolve to a different server.

# Expected behavior

Modoboa would show 'yellow' status when MX resolves to wrong server. 'green' status should only show when MX is configured and resolves to Modoboa instance.

# Video/Screenshot link (optional)
![screenshot_20170618_164329](https://user-images.githubusercontent.com/17307/27261126-4439bc16-5445-11e7-9c08-8d4fdb77d538.png)

**Comment by tonioo** [comment]
The MX check currently works as follows:
* If the domain has a MX record declared => green else red
* It is possible to restrict the allowed MX values. If the MX is not one of this values => red

Is it clearer?

**Comment by brylie** [comment]
Yes, but may still be misleading. In effect, if someone has misconfigured their MX record, they might still see a green label. Simply checking that MX is declared does not ensure that it is pointing to the Modoboa server, hence the recommendation for a warning tag if MX points to a different server. Just a thought.

**Comment by tonioo** [comment]
And if you define a list of allowed MX values through the admin panel (parameters) ?

**Comment by brylie** [comment]
That makes sense. Is there a way to do that by default, to assist new admin users?

**Comment by tonioo** [comment]
Not yet... I guess a configuration wizard would be the best but it can be long to develop. Maybe a documentation?

**Comment by brylie** [comment]
Yeah, good (inline or in-context) documentation is usually a win for end-users. :smiley_cat:

**Comment by tonioo** [comment]
@brylie Hi, where would you add this documentation?

**Comment by brylie** [comment]
Perhaps an MX section on the [Configuration page](http://modoboa.readthedocs.io/en/latest/configuration.html?highlight=mx)?

**Comment by almereyda** [comment]
I didn't expect this feature to exist. I think it will be good to even consider to split the developer and administrator (installation, configuration) documentation from the end user (instance administrators, domain administrators, simple users) documentation and highlight at this from an appropriate place.

![image](https://github.com/modoboa/modoboa/assets/1645308/ce2329e7-36f2-4b14-954e-f8bc22bfd83b)

The Spinx documentation is sometimes very technical to read, and not very appealing. A more visual approach for the latter audiences will greatly help in identifying such configuration options.

Eventually configuring this option should even be part of a default secure configuration example in the documentation of the Modoboa installer.

**Comment by Spitfireap** [comment]
Well I can assure you that is not fun to write Shpinx doc :). At least it is not suited for writing sweet documentation other than the technical one. 

I was writing a detailed guide on how to tune the installer and startup the deployment of an instance but it is on hold for now while we're working on the rspamd release. 

Please tell if you wish to contribute on writing such doc !

**Comment by almereyda** [comment]
I would like to contribute to writing end-user oriented documentation for Modoboa's four main audiences as seen from the frontend: Simple users, Domain administrators, Resellers and Super administrators.

I would even change the wording of these audiences a bit, and reorder them by degree of power in the New Account UI, with power-levels increasing to the bottom, and also in so Simple users are the default on top of the list:

- Users  
  No simple, as no other kind of type User is available.
- Domain administrator  
  People and power users understand this term intuitively.
- Reseller
- Instance/Site administrator  
  "Super" feels a bit elitist here

But don't allow me to overtake this issue. Should I open a new ticket for end-user documentation, and make some proposals (of technical documentation generators) how that could look like?

**Comment by Spitfireap** [comment]
You can open a Github discussion but the easiest way would be to come to the discord server. The link is available on the Readme ;).

**Comment by almereyda** [comment]
Discord, my anti-love. I'm coming.
